### ISSUE TITLE
When is "the one" comming to Craft 3 !?

### ISSUE THREAD

**Issue by H-i-red** [spicywebau/craft-neo#121]

**Comment by j2is** [comment]
+1 👍

**Comment by elfacht** [comment]
Srsly, please make it happen!

**Comment by timostsauber** [comment]
Is there a rough ETA?  Now that Craft3 is in RC

**Comment by maxwellhibbert** [comment]
I think his goal is Craft 3 launch

**Comment by timostsauber** [comment]
Have you heard when the Craft 3 might be moving from RC to final version?  I couldn't find any roadmap on their website.

**Comment by maxwellhibbert** [comment]
Mid April I think

**Comment by timostsauber** [comment]
Wow thats close innit. Thanks for the info maxwellhibbert!

**Comment by kisonay** [comment]
Would love to help test this, just no clue as to how to install the dev-3-port branch into my local env.

**Comment by kisonay** [comment]
Is there anything that we can do to help with this development? Only 2 months away from a GA.

**Comment by andrewhawkes** [comment]
Any idea on an ETA? Would be ace to use this on a new project we're doing on Craft 3. Would happily pay for this plugin on Craft 3!

**Comment by elfacht** [comment]
I already did the work and transferred my Neo fields to [Super Table](https://github.com/verbb/super-table) fields. Not as nice as Neo, but good for everyone who dislike waiting.

**Comment by nickdunn** [comment]
Looks like there are a couple of branches on the go, but no activity since January :-(

**Comment by andrewhawkes** [comment]
@nickdunn that's good to know. I don't think Super Table is going to work in this case so Neo on Craft 3 would be good.

**Comment by groe** [comment]
Craft 3 has been released today :) Any news yet?

**Comment by kisonay** [comment]
Ben updated the README on the dev-3-port branch. He is hoping to get Neo updated this week for Craft 3. https://github.com/benjamminf/craft-neo/commit/29f7b13966c40a4a0aa346a3886e46e99bc69031

**Comment by ejaedesign** [comment]
Also waiting for "The One" :)
I don't think the update has happened yet but hopefully soon.

**Comment by jarnstad** [comment]
Any information when the port is done on this brilliant plugin?

**Comment by waswebb** [comment]
*THE ONE* for CRAFT 3!

**Comment by kisonay** [comment]
I have tried the dev-3-port branch and although you can build fields with it, it still isn't in a state where you can save content in entries.

**Comment by lazlo-bonin** [comment]
Last update that mentioned being done in the week was more than 2 weeks ago now, any update from @benjamminf ? 😄 Waiting on Neo to migrate my website to Craft 3.

**Comment by jarnstad** [comment]
Has the craft 3 port stalled completely? Have a project that has been put on hold until neo for craft 3 is ready, so i would really like to see this in production soon.

**Comment by waswebb** [comment]
Please release Neo for Craft 3!
I have 2 projects with are waiting for Neo

**Comment by JonnyTurnerUK** [comment]
Doh! I thought this would have been one of the first to be ported. Need Neo. 😞

**Comment by mikethehud** [comment]
Newest commit message on the `dev-3-port` from last night: `Block structure is next, then almost ready`

**Comment by waswebb** [comment]
@benjamminf NEOO! 

I love your Plugin!

**Comment by kisonay** [comment]
dev 3 port branch is looking good! can't wait for this to be released.

**Comment by waswebb** [comment]
Is it ready to use?

**Comment by kisonay** [comment]
@waswebb I've been playing with it but it def isn't ready. I do find it usable enough to do some development though.

**Comment by lazlo-bonin** [comment]
Just a heads up for everyone else waiting, I reached out to Benjammin about the progress, and here's his reply (I hope he won't mind me posting!):

> Development is definitely still going for my plugins, just obviously a bit slowly. Unfortunately I can't give a definite timeline for Craft 3 support on my plugins, but I can tell you I'm about 75% done with Neo, so I'm predicting it'll be a week or two away if all goes to plan.
> 
> I feel your pain as we're currently starting the move to Craft 3, but Neo is the only thing holding us back. So it's quite important for us to have it completed in the coming weeks. Embedded Assets and No-Cache are the other plugins we use on every project, so they also need to be ported very soon as well (though they should be quick compared to Neo).
> 
> In case you're interested where Neo is currently is at, configuring Neo fields is complete, so pulling down a copy from `dev-3-port` will let you start architecting your sites. Saving fields is a work in progress, but is close to complete. After that the plugin will technically be usable, but I will need to create migrations from Craft 2 before it's officially released.
> 
> Sadly I can't guarantee a deadline for the ports, but they will be completed at some point in the near future. Thanks the kind words and patience.

tl;dr: It's definitely coming soon, we just have to be a little more patient!

**Comment by fThues** [comment]
We're trying to evaulate the plugin's current state, but after installing the `dev-3-port` branch via Composer, Craft doesn't the show the plugin in the Plugins section. Does this work for anyone?

**Comment by kisonay** [comment]
@fThues I’ve been using it without a problem? Are you sure it’s installing and that composer isn’t producing an error?

**Comment by fThues** [comment]
@kisonay Yeah, Composer does its thing and the plugin gets installed to `vendor/benjamminf/craft-neo`.

**Comment by waswebb** [comment]
No Update on Neo since 12 Days. Have anyone a idea why?

**Comment by jarnstad** [comment]
16 days and no updates. Is it possible to get an estimate when this plugin is released for craft 3? And a realistic estimate, until now all estimates have been completely wrong. Have a couple of projects for a customer that have stalled because I'm waiting for neo. Considering finding another solution instead of waiting.

**Comment by mwadera** [comment]
Any update?

**Comment by waswebb** [comment]
No. Till jet noting new :(

I have 5 projects on hold because of the missing kid Neo!

**Comment by BrandonJamesBorders** [comment]
So I figured I'd give the dev3 branch a go as I loved using it in Craft 2. So far, so good aside from a few Craft 2 styles being present. Here is a how to get it going. 

1. Download the [dev-3-branch](https://github.com/benjamminf/craft-neo/archive/dev-3-port.zip).
2. In your project root, create a folder named plugins.
3. Extract the contents from step 1 into a folder named Neo. Your folder structure should look like this:
- plugins
--neo
---client
---src
---.gitignore
--- . . . 
4. Open your composer.json file and add `"benjamminf/craft-neo": "^2.0"` to the `require` section. 
5. If your composer.json does not have a `repositories` section, create one to reference your new path. Here is what mine looks like: `"repositories": [
    {
      "type": "path",
      "url": "plugins/neo"
    }
  ],`
6. Run `composer update`.
7. Navigate Settings -> Plugins to instasll Neo. 
8. Ask @benjamminf how we can support him. (@benjamminf Any way we can support you?)

**Comment by kisonay** [comment]
@benjamminf I now things take time but could we get an update on this?

**Comment by mrksmts** [comment]
@BrandonJamesBorders Could you tell us something about how well is this version working also taking multiple sites into account?

@benjamminf please let us know if there is any way how we can support you.

**Comment by BrandonJamesBorders** [comment]
@mrksmts Using it in a multi-site environment now without major issue. There was one potential major oddity that I'm not sure if it was related to Craft or Neo where I had a channel available across all sites. The channel had one entry with dummy content in a Neo field. When I then disabled a channel for one of the sites, the non disabled entry blew up in form of a bunch of empty sub(?) Neo fields. Deleting the empty ones didn't help, I had to completely remove everything from the entry. 

Otherwise, all good!

**Comment by waswebb** [comment]
@benjamminf Can you please give us a status update? 
Here are so many people waiting for a Neo, please give us something.

**Comment by mrksmts** [comment]
@BrandonJamesBorders You could skip the first 3 steps of the guide. This also makes it easier to update once there is an update I think. 

1. Update the repositories section in composer.json:
```
"repositories": [
    {
      "type": "vcs",
      "url": "https://github.com/benjamminf/craft-neo.git"
    }
  ]
```

2. Run `composer require benjamminf/craft-neo dev-dev-3-port`

**Comment by jarnstad** [comment]
Anything? Really hope something will happen soon. Need this plugin so bad.

**Comment by smockensturm** [comment]
Anybody just update Craft to 3.0.11 and have the Neo dev plugin go all to hell?

**Comment by kisonay** [comment]
@smockensturm  I just upgraded and it seems fine to me. (The only issue I have had for a while now is that the checkbox next to each header/box is higher than it should be)

**Comment by ewhicher** [comment]
Any updates for us @benjamminf ? Sorry to bump 🤕

**Comment by jamielaughton** [comment]
Sorry to harass, but we're gonna have to ditch Neo in all our up-and-coming releases if there is no ETA announced for this update. We believe in you!

**Comment by groe** [comment]
Still hoping for a Craft 3 release, but seeing there are no commits for 2 months makes we wonder about the future of Neo. Are there any alternatives?

We're currently in the process of planning a new Craft site and with Craft 3 being released for 3 months starting a site from scratch with Craft 2.x becomes harder to justify every week :persevere:

**Comment by kisonay** [comment]
@groe  I have feared the same. I found that "Pimp My Matrix" has started development again and has been renamed to Spoon which may be an alternative.

**Comment by jamielaughton** [comment]
@groe @kisonay I agree, Pimp My Matrix looks like the way forward. Unhappy about losing the ability to re-use fields but even if Neo Craft 3 does come along there's no guarantee the plugin will be maintained in the future

**Comment by i-hardy** [comment]
@benjamminf You're still active on Github so I just hope you haven't muted this issue. Please could you at least confirm if you plan to continue development, if you welcome PRs for the Craft 3 version, if you would rather someone else take this on, etc.

I began work on a business-critical Craft-based marketing site in May, using the dev-3-port branch as you suggested above under the good faith assumption that development was ongoing on this plugin even if things were delayed. I need to be able to make a decision soon as to whether to continue with Neo or attempt to migrate over to another solution.

Is the Craft 3 port abandoned? Please can you just offer a yes or no answer. If it's a no then that's fine and I understand, but I really need to know one way or the other.

**Comment by jamielaughton** [comment]
Just noticed that dev-3-port branch was updated 2 days ago, hopefully we'll hear an update soon!

**Comment by chrisrowe** [comment]
It's happening...

**Comment by kisonay** [comment]
It is happening.... just slowly, not additional updates for the last 11 days.

**Comment by lazlo-bonin** [comment]
Is it possible to start architecting a site with the dev-3-port branch?
I don't need to use it in production yet, but I do need to start structuring my content.
If I start using it now, will I be able to migrate to the full version once it's ready?

**Comment by kisonay** [comment]
@lazio-bonin yes, the dev-3-port branch is usable (with a few bugs.) I have been using it for a few months but have started to look at other options since there is no indication on when it will be production ready and I have been holding a project because of it.

**Comment by kisonay** [comment]
@23d1 its mostly around third party field type plugins (so not really Neo's fault.) If you havne't experienced any issues, and are using it in production, that is great news.

**Comment by kisonay** [comment]
@23d1  have you had any issues with Live Preview and Neo? for some reason I am not getting the preview to update when modify the content with a Neo field.

**Comment by gregorydavidjenkins** [comment]
Live preview isn't working yet. I had a conversation with Ben a couple months ago and he said the structure is in place so as updates come they won't break the overall structure, but "bells and whistles" like live preview aren't set up yet.

**Comment by Osze** [comment]
I have more than 10 projects waiting for NEO Craft 3. Please, make it happen and for your sake, don't make it for free this time! I think we all want to contribute. :)

**Comment by divyan** [comment]
Any news on this?

**Comment by httpster** [comment]
In the early phases of a project where Neo would be a great addition, but if it's abandoned or is going to be awhile I will move in another direction. A progress update would be very helpful.

**Comment by lukebailey** [comment]
Looks like Ben has moved to Facebook (according to Spicy Web) and the repo has been moved to their organisation account - anyone their looking to maintain/release this?

**Comment by timeverts** [comment]
I've recently made contact with Tony Sambell, Director of Spicy Web, querying about whether Neo will still get Craft 3 support now that @benjamminf has left the company. 

Thankfully, the news sounds promising:

> We have taken over NEO and will have it Craft 3 ready asap. 
We use NEO on all our websites and now that we have ownership we are making it a high priority to get it C3 ready.

**Comment by ttempleton** [comment]
Hi everyone,

I'm Thomas, I'm a developer for Spicy Web, and I'll be involved in completing the Craft 3 version of Neo.  I've just updated the readme with a public message from Tony Sambell, the Director of Spicy Web, stating that Spicy Web will be supporting Neo from now on and intends to have the Craft 3 version completed and released by October 30.  We will, of course, endeavour to complete it sooner if possible, but October 30 is our absolute, must-have-it-done date.  I'll make sure to keep you all updated.  Thank you for your patience so far.

**Comment by 23d1** [comment]
Hey @ttempleton, I didn't want to open a whole new "issue" for a brief question;

Have you guys started looking into the new Project Config functionality in Craft 3.1, outlined [here](https://craftcms.com/news/craft-3-1-dev-preview)?

I'm sure it's on your radar, just curious—it seems like an amazing functionality that seems so logical it's crazy it wasn't built in into Craft 2. Anyway, looking forward to the release of Neo for Craft 3. Is the dev port still safe to use for testing purposes?

Cheers!

**Comment by ttempleton** [comment]
Hi @23d1

We've not yet looked into Project Config in detail, but we definitely agree, it's a fantastic new feature that will make life much easier.

Neo for Craft 3 is coming along well, it's still on track to be released by October 30, and I would say, for those who have been using it for testing purposes, that it's still safe to do so.  All of the structural work was completed by Ben before we took over, so nothing will change in that regard to make it less safe to use, and our work from here will involve implementation of necessary features and bugfixes.  I have live preview in a mostly-working state -- albeit still with a few issues that I hope to sort out early next week -- and then we'll move on to migrations from Craft 2.

As a side note: after we release the Craft 3 version, we do plan to return to the Craft 2 version and fix issues with the v1.5.0 release.

**Comment by theskyfloor** [comment]
@ttempleton great news... and exactly what i was hoping to hear as far as migrations go. are you going to make neo paid? i hope so, i want to be sure we can rely on it going forward :)

**Comment by ttempleton** [comment]
Hi @theskyfloor -- at this stage I can say it's likely that the Craft 3 version of Neo will be a paid plugin.  I can't yet confirm pricing, but will update with that information as soon as possible next week.

**Comment by theskyfloor** [comment]
@ttempleton awesome, that is fantastic news in my opinion!

**Comment by H-i-red** [comment]
Hi @ttempleton is there a way to install the dev-3 for previewing/testing ? Composer ?

**Comment by H-i-red** [comment]
Hi @timkelty , thanks but it´s not working "Your requirements could not be resolved to an installable set of packages", I have set "minimum-stability": "dev", "prefer-stable": true...  ?!

**Comment by ttempleton** [comment]
Hi @H-i-red

What @timkelty posted should work with one small change: Despite the changeover to Spicy Web, at this time the entry in the Composer project requirements should still be `benjamminf/craft-neo` (this will be updated in the near future and I will notify when it has been updated).

---

On another topic, regarding Neo pricing: My understanding is that the initial purchase price of Neo for Craft 3 will be very low, and the annual renewal will be a standard price.

**Comment by H-i-red** [comment]
Woohooo 🤘 **NEO** is back  💪 !! Thanks @ttempleton

**Comment by H-i-red** [comment]
I see the `duplicate` button is replaced by a `copy`. Here comes a future request.. 😬

1.  Be able to copy multiple selected `blocks` (Now I can only copy one at a time ?)
2. Bring back the `duplicate` button. For quick one copy use the `duplicate` button, for multiple, select your blocks and use the `copy`button. So you have two buttons one for `copy` & one for `duplicate`.

Found a 🐛 . Upon save the preview for the blocks is lost, returning `undefined` in `.preview-section`. When I expand/collapse the block it´s shown..

**Comment by 23d1** [comment]
I know this probably isn't the place for it, but I was thinking about making Neo a paid plugin... Being somewhat of an idealist in terms of open source and free software, I would urge that if there's a pricing model invoked—would it make sense to use a similar approach as ACF, where you essentially license the software, but can use it across many/all your projects?...

**Comment by theskyfloor** [comment]
@23d1 that's an interesting idea. Would love to hear SpicyWeb's thoughts on that... we do use Neo on a lot of projects and pricing like that would be beneficial to us - we would gladly pay a decent fee for it let's say 99 to 199 a year but be able to use it on any site we want.

**Comment by kisonay** [comment]
Interesting idea. If you go this route, Is it possible for two pricing levels? Single site and agency? Can’t forget about the little guys.

**Comment by luke-nehemedia** [comment]
I am not that much a fan of this idea ;)

I use neo on a lot of small budget projects or even for small charitable projects. All those sites/projects would not be able to pay such a high fee.

However, from an agency’s point of view it makes total sense.
Am 19. Okt. 2018, 20:53 +0200 schrieb kisonay <notifications@github.com>:

**Comment by 23d1** [comment]
Ideally it just remains free. I don't want the projects with micro budgets to have to use SuperTable or the likes. What was Ben's plans and wishes for this? Anyway, if it's not a huge fee I'd be fine with it, or developer licensing (one-time with low a low yearly cost for updates or something).

**Comment by tonysambell** [comment]
Hello All, Neo will indeed be becoming a paid plugin. However, we have given this much thought and the way we have decided to price it will allow us to prove it's worth to developers. From my current understanding, we cannot charge unlimited agency pricing via the Craft Plugin Store. But once you see how we have decided to price Neo this shouldn't be as much of a concern. I know we can never make everyone 100% happy but I am confident most of you will appreciate the method in our madness once we launch Neo.

**Comment by 23d1** [comment]
Totally get the choice of going toward a paid model. Personally, I'm super disappointed by it, however—as the ideal for me, and many I know, would be an agency pricing model. I guess it's time to look elsewhere. It was a great ride, though. Keep up the good work, and hopefully in the future there's room to have a couple of different licensing models.

Cheers! ✌🏻

**Comment by tonysambell** [comment]
Hello All, NEO is now officially in the Craft 3 plug-in store. We are very proud to see it finally Craft 3 ready. 
Yes, it is now a paid plug-in but Spicy Web as an agency will be giving NEO prompt and ongoing support.
